app/controllers/intercom_integrations_controller.rb

```diff
@@ -0,0 +1,102 @@
+
+    Rails.logger.info("Loading intercom accounts for company #{company.name}...")
+    accounts = company.accounts
+    account_identifiers = accounts.map(&:source_identifier).to_set
+    companies_to_add = intercom.companies.scroll.reject { |c| account_identifiers.include?(c.name) }
+    Account.transaction do
+      num_companies_saved = companies_to_add.map { |c| convert_to_account_and_save(c) }.sum
+      Rails.logger.info("Attempted to save #{companies_to_add.length} companies. Saved #{num_companies_saved} companies")
+    end
+    render json: { success: true }
+  end
+
+  def load_customers
+    company, intercom = extract_company_and_intercom_client
+    return unless intercom
+
+    Rails.logger.info("Loading intercom customers for company #{company.name}...")
+    existing_user_emails = User.all.map(&:email).to_set
+    users_to_add = intercom.contacts.all.reject { |c| existing_user_emails.include?(c.email) }
+    User.transaction do
+      num_users_saved = users_to_add.map { |u| create_user(u) }.sum
+      Rails.logger.info("Attempted to save #{users_to_add.length} users. Saved #{num_users_saved} users")
+    end
+    render json: { success: true }
+  end
+
+  def load_membership
+    company, intercom = extract_company_and_intercom_client
+    return unless intercom
+
+    Rails.logger.info("Setting account and customer membership for company #{company.name}...")
+    all_user_hash = User.all.map { |u| [u.email, u] }.to_h
+
+    # Setup account membership
+    valid_contacts = intercom.contacts.all.select { |c| all_user_hash.include?(c.email) }
+    AccountMembership.transaction do
+      num_account_memberships_added = valid_contacts.map { |c| add_account_membership_for_user(c, all_user_hash[c.email]) }.sum
+      Rails.logger.info("Attempted to create account memberships for #{valid_contacts.length} users. Saved #{num_account_memberships_added} account memberships")
+    end
+
+    # Setup customer membership
+    existing_customer_emails = company.customers.map(&:email).to_set
+    # Check if we haven't added it yet and if the user exists
+    customers_to_add = intercom.contacts.all.select { |c| !existing_customer_emails.include?(c.email) && all_user_hash.include?(c.email) }
+    Membership.transaction do
+      num_memberships_added = customers_to_add.map { |c| add_user_as_customer(company, all_user_hash[c.email]) }.sum
+      Rails.logger.info("Attempted to create #{customers_to_add.length} customer memberships. Saved #{num_memberships_added} memberships")
+    end
+    render json: { success: true }
+  end
+
+  def extract_company_and_intercom_client
+    company = Company.find(params[:company_id])
+    intercom_access_token = company.company_setting.intercom_access_token
+    intercom = nil
+    intercom = Intercom::Client.new(token: intercom_access_token) if intercom_access_token
+    [company, intercom]
+  end
+
+  def convert_to_account_and_save(intercom_company)
+    a = current_company.accounts.new
```

Comment:
Forgive me for shouting code styling 🙈, but nice to name variables for readability 

```suggestion
    account = current_company.accounts.new
```